Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -373,13 +372,38 @@ public bool Equals(string? other)
             int i = 0;
 
             return
-                TestInt(_i1) &&
-                TestInt(_i2) &&
-                TestInt(_i3) &&
-                TestInt(_i4) &&
-                TestInt(_i5);
+                TestInt64(_i1) &&
+                TestInt64(_i2) &&
+                TestInt32(_i3);
+
+            bool TestInt64(ulong k)
+            {
+                return
+                    TestDigit(k >> 60) &&
+                    TestDigit((k >> 56) & 0xF) &&
+                    TestDigit((k >> 52) & 0xF) &&
+                    TestDigit((k >> 48) & 0xF) &&
+                    TestDigit((k >> 44) & 0xF) &&
+                    TestDigit((k >> 40) & 0xF) &&
+                    TestDigit((k >> 36) & 0xF) &&
+                    TestDigit((k >> 32) & 0xF) &&
+                    TestDigit((k >> 28) & 0xF) &&
+                    TestDigit((k >> 24) & 0xF) &&
+                    TestDigit((k >> 20) & 0xF) &&
+                    TestDigit((k >> 16) & 0xF) &&
+                    TestDigit((k >> 12) & 0xF) &&
+                    TestDigit((k >> 8) & 0xF) &&
+                    TestDigit((k >> 4) & 0xF) &&
+                    TestDigit(k & 0xF);
+
+                bool TestDigit(ulong j)
+                {
+                    char c = j < 10 ? (char)('0' + j) : (char)(j + 0x57);
+                    return other[i++] == c;
+                }
+            }
 
-            bool TestInt(uint k)
+            bool TestInt32(uint k)
```

Comment:
ditto

Reply:
As #11222 was accepted, I will remove this code